py-polars/src/lazyframe/mod.rs

```diff
@@ -65,17 +65,37 @@ impl PyLazyFrame {
         }
     }
 
-    #[cfg(all(feature = "json", feature = "serde_json"))]
-    fn serialize(&self, py_f: PyObject) -> PyResult<()> {
+    /// Serialize into binary data.
+    fn serialize_binary(&self, py_f: PyObject) -> PyResult<()> {
+        let file = BufWriter::new(get_file_like(py_f, true)?);
+        ciborium::into_writer(&self.ldf.logical_plan, file)
+            .map_err(|err| PyValueError::new_err(format!("{err:?}")))?;
+        Ok(())
+    }
+
+    /// Serialize into a JSON string.
+    #[cfg(feature = "json")]
+    fn serialize_json(&self, py_f: PyObject) -> PyResult<()> {
         let file = BufWriter::new(get_file_like(py_f, true)?);
         serde_json::to_writer(file, &self.ldf.logical_plan)
             .map_err(|err| PyValueError::new_err(format!("{err:?}")))?;
         Ok(())
     }
 
+    /// Deserialize a file-like object containing binary data into a LazyFrame.
+    #[staticmethod]
+    fn deserialize_binary(py_f: PyObject) -> PyResult<Self> {
+        let file = get_file_like(py_f, false)?;
+        let reader = BufReader::new(file);
+        let lp = ciborium::from_reader::<DslPlan, _>(reader)
```

Comment:
ciborium also offers a `from_reader_with_buffer` which allows finetuning the size of the buffer (default is 4K). I don't really know what size data we're usually looking at, so leaving this at default for now.